addons/web_editor/static/src/js/backend/convert_inline.js

```diff
@@ -634,20 +634,30 @@ async function toInline($editable, cssRules, $iframe) {
 async function flattenBackgroundImages(editable) {
     for (const backgroundImage of editable.querySelectorAll('*[style*=background-image]')) {
         if (backgroundImage.parentElement) { // If the image was nested, we removed it already.
-            const canvas = await html2canvas(backgroundImage);
+            const iframe = document.createElement('iframe');
+            const style = getComputedStyle(backgroundImage);
+            const clonedBackground = backgroundImage.cloneNode(true);
+            clonedBackground.style.height = style['height'];
+            clonedBackground.style.width = style['width'];
+            iframe.style.height = style['height'];
+            iframe.style.width = style['width'];
+            backgroundImage.after(iframe);
+            iframe.contentDocument.body.append(clonedBackground);
+            iframe.contentDocument.body.style.margin = 0;
```

Comment:
The commit message is explicit but I think this warrants a comment in the code as well.
I suggest the following (break at 80 chars):
```
// The html2cans library clones the entire document of the provided node. That is a slow but necessary process in order to fetch all the styles that apply to that node. However, it is not needed in our case because all the styles have already been inlined on the node itself at this point.
```

Note that I thought about a case where this would not work. Something like this:
```html
<div style="text-color:red">
    A
    <div style="background-image: ...">B</div>
    C
</div>
```
In that case B appears red on the screen but will end up black in the result from `html2canvas`. I discussed with @Zinston about it and we do not think that it is possible to create a case like this in Odoo with the current possibilities of the editor, but it is a limitation of this system nonetheless, as it might become possible in the future.

Reply:
We could  wrap the target element with all the computed properties of the parent of the target elements. Do you want me to make that change?

Reply:
Well, if we apply the GA's teachings, since we do not think this case can happen in Odoo, we should not add more code to support it _in advance_. We'll do it when we have an actual case that requires it.